%!TEX root = ../thesis.tex
% TODO: Change this?

\chapter{The \textit{IgH} locus in \textit{Nothobranchius furzeri}}  
\onehalfspacing

{\LARGE First Draft, \today}

\renewcommand{\floatpagefraction}{.85}%
\renewcommand{\topfraction}{.85}%

%\section*{Summary}
%
%
%The turquoise killifish IgH locus resembles that of medaka, the most closely-related teleost species with a characterised \textit{IgH} locus, in several important respects, including an unusual IgM-TM splice pattern different from most teleosts and the absence of the teleost-specific antibody isotype IgZ/T. The shared absence of IgZ/T in these related species was particularly striking, as all previously-characterised teleost loci except medaka and grasscarp have been found to possess this isotype. Given their relatedness, it was hypothesised that \dots
\pagebreak

% Sections

\section{Introduction} % TODO: Cite this properly

The native structure of the immunoglobulin heavy chain (\igh{}) locus determines the state space of antibody heavy-chain diversity in a species, including the range of \vh, \dh and \jh segment choices available in VDJ recombination, the relationship between VDJ recombination and isotype choice, and the ability of processes such as segment inversion, gene conversion and class-switch recombination to affect the diversity and functionality of the antibody repertoire. The diversity produced by VDJ recombination, junctional diversity and secondary diversification processes in this locus are responsible for the majority of variation in antigen-specificity within a B-cell population, while the choice of isotype among the available \igh{} constant regions determines the antibody's effector function and relationship with the rest of the immune system. Understanding the native structure of the \igh{} locus is therefore essential for understanding how the adaptive immune system functions in an organism, while comparing loci between species enables the evolutionary history of adaptive immunity across lineages to be analysed, providing crucial insight into the complex history of this essential biological system. Last but not least, by providing thorough documentation of the \igh{} gene segments present in a species, characterising the \igh{} locus in a species is an essential forerunner to quantitative analysis of adaptive immunity using immunoglobulin sequencing. % Too long?

Previous work has characterised \igh{} locus structure in a number of teleost species, including zebrafish, medaka, stickleback, rainbow trout, fugu, and several salmonid species. This work has revealed striking diversity in the size, structure and functionality of teleost \igh{} loci. However, the number of loci characterised is very small compared to the evolutionary diversity of known teleost species, and is mainly confined to major aquaculture species (trout, catfish, salmon) or research models (zebrafish, stickleback, medaka), with characterised species often quite distantly related to one another across the teleost evolutionary tree. This relatively sparse sampling of teleost \igh{} loci has left wide swathes of teleost diversity without any characterised \igh{} loci, and has prevented higher-resolution analysis of locus structural evolution across closely related species.

In this chapter, therefore, I present complete characterisations of the \igh{} loci of two important model organisms from the Cyprinodontiformes, a diverse clade of primarily freshwater teleost fish. \nfu, the turquoise killifish, has recently emerged as an important model system for ageing research, % TODO: Add link to intro section on TK: (see \Cref{sec:intro-killifish})
and is also of evolutionary and ecological interest due to its short lifespan, extreme natural environment, and unusual life history. The southern platyfish \xma, meanwhile, is an important model organism in evolutionary ecology and population genetics. %, and \dots . % TODO: More about Xiphophorus here, plus citations
Comparing the \igh{} loci of these two closely-related species reveals dramatic and unexpected differences in immune structure, which when combined with previously-published loci from related species suggest unexpected patterns of locus evolution within this group of teleost fishes: most strikingly, the specialised mucosal antibody isoform \igh{Z} appears to have been independently lost in multiple closely-related lineages. 

To further investigate this and other surprising features of \igh{} locus evolution in this lineage, I performed a partial reconstruction and analysis of the \igh{} loci from ten further cyprinodontiform species (\Cref{fig:species-tree-large-taxa}), as well as from a new and improved genome assembly of medaka (\textit{Oryzias latipes}). Phylogenetic analysis of these data confirms the repeated duplication and loss of \igh{Z} in this lineage and provides evidence for ????. % TODO: fill this in after completing multispecies analysis section
Taken together, this analysis significantly extends our knowledge of \igh{} locus and especially constant-region diversity in teleost fish, and establishes the cyprinodontiforms, and especially the African killifishes, as a highly promising collection of model systems for comparative evolutionary immunology.

\begin{figure}
	\centering
	\includegraphics[width=0.9\textwidth]{_Figures/png/species-tree-large-taxa}
	\caption[Cladogram of analysed species]{\textbf{Cladogram of species included in this analysis.} Boldface type indicates species for which new, complete \igh{} locus assemblies were generated for this study; other species were either previously-characterised reference species (\species{G. aculeatus}, \species{O. latipes}) or underwent constant-region characterisation only (all other species). Labelled vertical bars designate higher taxa of interest.}
	\label{fig:species-tree-large-taxa}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% KILLIFISH LOCUS SECTION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\input{B_Chapters/3_Locus/Sections/Killifish}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% KILLIFISH LOCUS SECTION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\input{B_Chapters/3_Locus/Sections/Platyfish}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% COMPARATIVE SECTION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\input{B_Chapters/3_Locus/Sections/Comparative}