%!TEX root = ../thesis.tex
% TODO: Change this?

\chapter{The \textit{IgH} locus in \textit{Nothobranchius furzeri}}  
\onehalfspacing

% Chapter summary (should fit on title page)

\input{B_Chapters/3_Locus/Sections/_Summary}

\pagebreak

% Sections

\definecolor{codegreen}{rgb}{0,0.6,0}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolour}{rgb}{0.95,0.95,0.92}

\lstset{
    backgroundcolor=\color{backcolour},   
    commentstyle=\color{codegreen},
	basicstyle=\ttfamily,
	columns=fixed,
	%frame=single,
    breaklines=true,
    breakatwhitespace=true,
    postbreak=\mbox{\textcolor{red}{$\hookrightarrow$}\space}, % TODO: Change linebreak character?
    language=bash,
    keepspaces=true,
    aboveskip=0.6em,
    belowskip=0.3em,
    }
	

\section{Introduction}

The structure of the immunoglobulin heavy chain (\textit{IGH}) locus defines the state space of antibody heavy-chain diversity and function in a species, determining both the V, D and J segments available for VDJ recombination and the range, structure and function of the possible antibody classes available to that organism. In addition to determining the possible antibody effector functions in a species, this locus is also responsible for the majority of diversity in antigen-specificity available to an organism, due to the exceptionally high diversity of the CDR3 junctional region compared to that of light chain loci. 

Determining the sequence and structure of the IGH locus in a species is therefore an essential step towards understanding and manipulating humoral adaptive immunity in a species, both in its own right and as an important stepping stone towards successful immunoglobulin sequencing in that taxon.

In addition to the instrumental importance of the IGH locus structure for immunoglobulin sequencing and other experimental interventions, however, it is also an important and interesting topic in its own right, revealing much about the functionality and evolution of antibody heavy chains in and between related species. In particular, while a number of IGH loci have been sequenced and characterised in the teleost fishes, these loci have been taken from a wide range of relatively distantly related species, chosen due to their use in agriculture or their prior status as experimental model organisms. There is therefore a lack of more closely-related loci that can be used to investigate the evolution of humoral immunity at higher temporal resolution. 

In this chapter, therefore, I present detailed characterisation of the IGH locus in the turquoise killifish, as well as that of another important model organism in the Cyprinodontiformes, namely the southern platyfish \textit{Xiphophorus maculatus}. In addition, I provide partial characterisations of loci from ten other cyprinodontiform species, with a focus on the constant-region exons and isotypes present across this important and diverse teleost clade. The results of these investigations demonstrate rapid evolution in the core functional components of humoral adaptive immunity within the cyprinodontiforms, with multiple independent losses of the IGHZ isotype and [something about CM and CD here]. These results raise exciting questions about the kinetics of immune-system evolution in this taxon and indicate the value of the cyprinodontiforms, and the African killifishes in particular, as a model for comparative evolutionary immunology in teleost fishes. 

\section{Materials and methods} % Maybe, maybe not, depends on final structure - maybe put relevant stuff here for draft and then move it later?

\subsection{General data processing and pipeline structure}

Unless otherwise specified, processing and analysis of biological data was performed using standard \lstinline[language=R]{Bioconductor} \citep{huber2015bioconductor} packages: \lstinline[language=R]{Biostrings} \citep{pages2017biostrings} and \lstinline[language=R]{BSgenome} \citep{pages2018bsgenome} for biological sequence data, \lstinline[language=R]{GenomicRanges} \citep{lawrence2013genomicranges} for sequence ranges, and \lstinline[language=R]{genbankr} \citep{becker2018genbankr} and \lstinline[language=R]{rentrez} \citep{winter2017rentrez} for GenBank files.

Processing of tabular data was performed using the Tidyverse suite of tools, especially \lstinline[language=R]{readr} \citep{wickham2018readr}, \lstinline[language=R]{dplyr} \citep{wickham2018dplyr}, \lstinline[language=R]{tidyr} \citep{wickham2018tidyr} and \lstinline[language=R]{stringr} \citep{wickham2018stringr}. Snakemake \citep{koster2012snakemake} was used to design and run data-processing pipelines.

\subsection{Collating reference sequences}

In order to identify putative \textit{IGH} loci and gene segments in a new genome, it is important to have well-curated reference datasets from already-characterised species. However, most publications presenting characterisations of \textit{IGH} loci do not provide easy-to-use databases of trimmed and curated gene segments. In order to obtain such segments, further analysis was performed on publically-available data from three reference species with previously-characterised \textit{IGH} loci: medaka (\textit{Oryzias latipes}), zebrafish (\textit{Danio rerio}) and three-spined stickleback \textit{Gasterosteus aculeatus}), as described below. Following automatic sequence extraction, the reference sequences were checked manually for any severely pathological (e.g. out-of-frame) sequences and corrected before being used for inference in novel loci.

\subsubsection{Medaka}
\label{sec:ref_locus_ola}

Genbank files of the annotated medaka \textit{IgH} locus were downloaded from the supplementary information of the medaka locus paper (\citep{magadan2011medaka}, additional file 6) and corrected to make them parsable by \texttt{genbankr}. Locus sequence and annotation ranges were extracted from these Genbank files into FASTA and TSV formats, respectively, and segment annotations were renamed to match the naming conventions used in other species. VH, DH, JH and constant-region exon nucleotide sequences were extracted from the locus sequence using these annotations. Amino-acid sequences for VH, JH and constant-region sequences were obtained automatically by identifying the reading frames which minimised the number of STOP codons in each sequence.

\subsubsection{Stickleback}
\label{sec:ref_locus_gac}

Limited sequence information on the IgH locus in stickleback, including VH segments and bulk (non-exon-separated) constant regions was provided in a GenBank file in the locus characterisation paper for medaka (\citep{magadan2011medaka}, additional file 6), while additional sequence information (including DH and JH nucleic-acid sequences and amino-acid sequences of constant-region exons) was extracted manually from one of the stickleback locus papers (\citep{bao2010stickleback},  Figure S1 to S4) into FASTA files. As with medaka, the Genbank reference file was downloaded, corrected and parsed to yield a FASTA file of the locus sequence and TSV files of annotation ranges. VH sequences were extracted from the locus sequence using these annotation ranges and translated as specified in \Cref{sec:ref_locus_ola}; JH sequences provided by \citep{bao2010stickleback} were translated such that the final nucleotide formed the last position of the final codon.

To obtain nucleic-acid sequences of the constant-region exons, the amino-acid sequences from \citep{bao2010stickleback} were aligned to the locus sequence with \lstinline{TBLASTN} \citep{gertz2006tblastn}, with a query coverage threshold of 40\% and a maximum of three HSPs per query sequence:

\begin{lstlisting}[language=bash]
tblastn -query <ch_aa_fasta> -subject <gac_locus_fasta> -qcov_hsp_perc 40 -max_hsps 3 -outfmt '<output_format>' > <output_path>
\end{lstlisting}

\noindent with the following standardised tabular output format: 

\begin{lstlisting}
6 qseqid sseqid pident qcovhsp length mismatch gapopen gaps sstrand qstart qend sstart send evalue bitscore qlen slen
\end{lstlisting}

To filter out alignments across subloci, any alignment of an exon upstream of the annotated boundaries of its corresponding bulk constant region (whose ranges were specified in the Genbank file) was discarded; the alignment with the highest score for each exon was then used to extract the corresponding nucleic-acid sequence from the locus. In order to control for any errors, either during manual extraction of locus sequences from the paper or in the paper itself, these nucleic-acid sequences were then re-translated to generate new amino-acid sequences, again using the translation frame producing the fewest STOP codons; these sequences were then used in place of the reference files in downstream applications.

\subsubsection{Zebrafish}
\label{sec:ref_locus_dre}

Genbank files corresponding to the zebrafish \textit{IgH} locus were provided (without segment annotations) on Genbank by \citep{danilova2005zebrafish}; this publication also provided detailed co-ordinates for the VH, DH and JH segments (but not constant exons) on these sequences. Aligned amino-acid sequences were provided for the exons of IGHM and IGHZ, but no detailed information about IGHD exons could be found for these sequences; as a result, reference information about C$\delta$ exons was not used from this species.

As with stickleback, the amino-acid sequences provided were aligned to the locus sequences  with \lstinline{TBLASTN} to identify and extract exon nucleic-acid sequences, which were then translated using the frame yielding the fewest STOP codons for each sequence. VH sequences were obtained using the ranges provided in \citep{danilova2005zebrafish} and translated in the same manner. DH and JH nucleotide sequences were obtained directly from \citep{danilova2005zebrafish}; as with stickleback, JH amino-acid sequences were obtained by translating the nucleotide sequences in the frame such that the final nucleotide formed the last position of the final codon.

\subsection{Identifying putative locus sequences}
\label{sec:locus_scaffolds}

In order to identify sequences potentially containing part of an IGH locus, reference VH, JH and constant-region nucleotide and amino-acid sequences from medaka, stickleback and zebrafish were mapped to genome assemblies from \textit{N. furzeri}, \textit{X. maculatus} and ten other cyprinodontiform species (\Cref{tab:cyprinodontiform_genomes}) using BLAST \citep{altschul1990blast,altschul1997blast}. Nucleotide sequences were aligned to the locus using the relatively permissive \lstinline{blastn} algorithm (as opposed to e.g. \lstinline{megablast} or \lstinline{dc-megablast}):

\begin{lstlisting}
blastn -tastk blastn -query <reference_exon_fasta> -subject <locus_fasta> -outfmt "<output_format>"
\end{lstlisting}

Protein sequences, meanwhile, were aligned using the standard \lstinline{blastp} algorithm:

\begin{lstlisting}
blastp -query <reference_exon_fasta> -subject <locus_fasta> -outfmt '<output_format>'
\end{lstlisting}

In both cases, the same tabular output format specified in \Cref{sec:ref_locus_gac} was used, to provide a predictable format for downstream processing of BLAST alignment tables.

Following alignment of reference sequences, overlapping alignments to reference segments of the same segment type, isotype (if applicable) and exon number (if applicable) were collapsed together, keeping track of the number of collapsed alignments and the best E-values and bitscores obtained for each alignment group. Alignment groups with a very poor maximum E-value ($> 0.001$) were discarded, as were groups consisting of fewer than two alignments and groups overlapping with a much better alignments to a different sequence type, where ``much better" was defined as a bitscore difference of at least 33. Following resolution of conflicts, VH and CH alignments underwent a second filtering step of increased stringency, requiring a minimum E-value of $10^{-10}$ to be retained. 

Following alignment filtering, scaffolds containing surviving alignments to at least two distinct segment types (where VH, JH, and each type of constant-region exon each counted as one segment type), or alignments to one segment type covering at least 1\% of the scaffold's total length were retained as potential locus scaffolds. These candidate scaffolds were subjected to more precise downstream analysis as appropriate to that species: \textit{N. furzeri} and \textit{X. maculatus} scaffolds were used as input to a more thorough locus assembly process (\Cref{sec:locus_assembly_methods}), while scaffolds from other species were searched for constant-region exons without further locus assembly (\Cref{sec:ch_char_crossspecies}).

\begin{table}
\centering
\caption{Genome assemblies used to identify putative \textit{IGH} locus sequences in cyprinodontiform fishes}
\begin{threeparttable}
\begin{tabular}{>{\itshape}l>{\itshape}llc}\toprule
\textnormal{\textbf{Genus}} & \textnormal{\textbf{Species}} & \textbf{Common Name} & \textbf{GenBank Assembly Accession}\\\midrule
Nothobranchius & furzeri & Turquoise killifish & NA\tnote{1}\\\midrule
Xiphophorus & maculatus & Southern platyfish & GCA\_000241075.1\\
Austrofundulus & limnaeus & -- & GCA\_001266775.1\\
Fundulus & heteroclitus & Mummichog & GCA\_000826765.1\\
Poecilia & formosa & Amazon molly & GCA\_000485575.1\\
Poecilia & reticulata & Guppy & GCA\_000633615.1\\
Cyprinodon & variegatus & Sheepshead minnow & GCA\_000732505.1\\
Kryptolebias & marmoratus & Mangrove rivulus & GCA\_001649575.1\\\midrule
Aphyosemion & australe & Lyretail panchax & NA\tnote{2}\\
Callopanchax & toddi & -- & NA\tnote{2}\\
Pachypanchax & playfairii & Golden panchax & NA\tnote{2}\\
Nothobranchius & orthonotus & Spotted killifish & NA\tnote{2}\\
\bottomrule\end{tabular}
	\begin{tablenotes}
	\item[1] Willemsen \textit{et al.}, unpublished at time of writing
	\item[2] Cui \textit{et al.}, unpublished at time of writing
	\end{tablenotes} % TODO: Update this once genomes are uploaded
\end{threeparttable}
\label{tab:cyprinodontiform_genomes}
\end{table}

\subsection{Assembling the \textit{N. furzeri} and \textit{X. maculatus} \textit{IGH} loci}
\label{sec:locus_assembly_methods}

\subsection{Characterising locus segments in \textit{N. furzeri} and \textit{X. maculatus}}

To characterise constant and variable gene segments on newly-assembled \textit{IGH} loci, databases of segment sequences from previously-characterised loci were used as reference databases. For \textit{N. furzeri}, the stickleback, medaka and zebrafish loci were used as references; for \textit{X. maculatus}, these three species plus \textit{N. furzeri} were used. In all cases, the analyses were repeated using the newly-characterised additional segments as an additional reference, but no additional segments were discovered.

\subsubsection{VH}

To identify VH segments on newly characterised loci, reference VH segments were used to construct a multiple-sequence alignment with \lstinline{PRANK} \citep{loytynoja2014prank}:

\begin{lstlisting}[language=bash]
prank -d=<reference_vh_db> -o=<output_path> -gaprate=0.00001 -gapext=0.00001 -F -termgap
\end{lstlisting}

The resulting alignment was used as an input to \lstinline{NHMMER} \citep{wheeler2013nhmmer,eddy2011hmm,eddy2009homology,eddy2008alignment}, which constructs a Hidden Markov Model from a multiple-sequence alignment and uses it to identify matching sequences in a reference sequence:

\begin{lstlisting}[language=bash]
nhmmer --dna --notextw --tblout <output_path> -T 80 <vh_alignment> <locus_sequence_path>
\end{lstlisting}

The resulting match table was used to identify candidate ranges in the locus sequence corresponding to VH segments; these ranges were extended by 9bp at either end to account for boundary errors, and the corresponding nucleotide sequences were extracted to a FASTA file. Each sequence was then checked and refined manually: 3' ends were identified by the start of the RSS heptamer sequence (typically \texttt{CACAGTG}), if present, while 5' ends and FR/CDR boundaries were identified using IMGT/DomainGapAlign \citep{ehrenmann2011domaingapalign} with the default settings. Where necessary, IMGT/DomainGapAlign was also used to IMGT-gap the VH segments in accordance with the IMGT unique numbering \citep{lefranc2003vnumbering}. % TODO: Cite something for RSS consenses; cite standard bioconductor packages for sequence/range manipulation etc 

An initial amino-acid sequence for each VH segment was produced automatically from the extracted nucleotide sequence by identifying the reading frame which minimised the number of STOP codons in the sequence; this worked well for most segments. VH amino-acid sequences were then refined (and in a few cases re-translated) from the manually-refined nucleotide sequences, including end-refinement and FR/CDR boundary identification.

Following extraction and manual curation, VH segments were grouped into families based on their pairwise sequence identity. In order to assign segments to families, the nucleotide sequence of each VH segment in a locus was aligned to each other segment using Needleman-Wunsch global alignment, as implemented in the \lstinline{pairwiseAlignment} function from the \lstinline{R} package \lstinline{Biostrings}. The percentage sequence identity was computed using the \lstinline{pid} function from the same package, with percentage identity defined as given in \Cref{eq:pid}; the resulting identity matrix was used to perform single-linkage heirarchical clustering on the VH segments, and the resulting dendrogram was cut at 80\% sequence identity to obtain VH families. These families were then numbered based on the order of the first-occurring VH segment from that family in the first-occurring sublocus in the parent locus, and each VH segment was named based on its parent sublocus, its family, and its order among elements of that family in that sublocus. %TODO: Citation for VH families

\begin{equation}
\mathrm{\%~sequence~identity} = 100 \times \frac{\mathrm{\#~identical~positions}}{\mathrm{\#~total~aligned~positions}}
\label{eq:pid}
\end{equation}

\subsubsection{JH}

As with VH segments, JH segments were identified by building a multiple-sequence alignment with \texttt{PRANK} and using it to construct an HMM with \lstinline{nhmmer}; the parameters used were the same as for VH segments, except that there was no minimum score for \lstinline{nhmmer} to report a sequence match (\lstinline{-T 0} instead of \lstinline{-T 80}). The resulting sequence ranges were extended by 20bp on either end and extracted into FASTA format. These sequences were then trimmed automatically by identifying the RSS heptamer sequence at the 5' end and the splice junction motif (\texttt{GTA}) at the 3' end. The JH nucleotide sequences were then checked and refined manually.

\begin{wraptable}{r}{5.5cm}
\caption{Regex patterns used to search for conserved W118 residues in JH sequences during AUX file generation}\label{tab:jh-aux-patterns}
\begin{tabular}{r>{\ttseries}l}\toprule  
\# & Pattern \\\midrule
1 & TGGGBNNNNGBN\\
2 & TGGGBNNNGBN\\
3 & TGGGBNNNNNGBN\\
4 & TGGGBNNNNNNGBN\\
5 & TGGGBN\\\bottomrule
\end{tabular}
\end{wraptable}

\lstinline{IgBLAST} \citep{ye2013igblast} identifies CDR3 boundaries for recombined IGH VDJ sequences using an AUX file specifying the reading frame of each JH segment, along with the co-ordinate of the conserved \texttt{TGG} codon (corresponding to the conserved W118 residue in the recombined sequence \citep{lefranc2014immunoglobulins}) marking the CDR3/FR4 boundary. An AUX file for the inferred JH segments was generated automatically by searching for the conserved sequence using a series of regular-expression patterns of decreasing stringency (\Cref{tab:jh-aux-patterns}), taking the first match in each sequence as the desired residue; this determined both the reading frame and the W118 sequence co-ordinate. Once generated, the AUX file was then used to determine the reading frame for automatically translated the JH sequences; both the AUX file and amino-acid FASTA file were then edited to incorporate any manual refinements made to the JH nucleotide sequences.

Curated JH sequences were named based on their order within their parent sublocus and, where applicable, on whether they were upstream of IGHZ or IGHM constant regions. 

\subsubsection{DH}

Unlike VH and JH gene segments, DH segments are too short and variable to be found effectively using an HMM-based search strategy. Instead, DH segments in assembled loci were located using their distinctive pattern of flanking recombination signal sequences: an antisense RSS in 5', then a short D-segment, then a sense RSS in 3'. Potential matches to this pattern were searched for using \lstinline{FUZZNUC} from the EMBOSS collection of bioinformatics tools \citep{rice2000emboss}, with a high error tolerance to account for deviations from the conserved sequence in either or both of the RSSs:

\begin{lstlisting}
fuzznuc -pattern 'GGTTTTTGTN(10,14)CACTGTGN(1,25)CACAGTGN(10,14)ACAAAAACC' -pmismatch 8 -rformat gff -outfile <output_path> <locus_sequence_path>
\end{lstlisting}

This generated a GFF file \citep{stein2010generic} of permissive matches, representing potential DH segments; these were then grouped by sequence co-ordinate, and higher-mismatch candidates overlapping with a lower-mismatch alternative were discarded.

Orientation of DH segments based on their own sequence is challenging, as the segments themselves have no clear conserved structure and the flanking RSSs are rotationally symmetric. To overcome this problem and orientate the DH segments on the locus, the table of DH candidate ranges was combined with previously-identified (and easier to orientate) VH and JH ranges. Each DH candidate was then orientated based on the orientations of its flanking segments: segments with an oriented segment immediately upstream or downstream adopted the orientation of that segment, while segments with contradictory orientation information were discarded. This process was repeated until all DH segments had either been orientated or discarded.

After orientation, the DH ranges were used to extract DH sequences in FASTA format from the locus sequence; these sequences then underwent a second, more stringent filtering step, in which sequences lacking the most conserved positions in each RSS were discarded \citep{grep}:

\begin{lstlisting}
grep -B 1 '[ACTG]\{{25,27\}}TG[ACTG]\{{1,25\}}CA[ACTG]\{{25,27\}}' <dh_fasta> | sed '/^--$'/d > <output_fasta>
\end{lstlisting}

Finally, the identified DH candidates were checked manually, candidates without good RSS sequences were discarded, and flanking RSS sequences were trimmed to obtain DH segment sequences. As with JH, these were numbered based on their order within their parent sublocus and, when applicable, on whether they were upstream of IGHZ or IGHM constant regions.

\subsubsection{CH}
\label{sec:locus_char_ch}

To detect and identify constant-region exons in the characterised loci, constant-region nucleotide and protein sequences from reference species were mapped to the locus sequence using BLAST \citep{altschul1990blast,altschul1997blast}, in the same manner described for putative locus scaffolds in \Cref{sec:locus_scaffolds}.
Following alignment of reference sequences, overlapping alignments to reference segments of the same isotype and exon number were collapsed together, keeping track of the number of collapsed alignments and the best E-values and bitscores obtained for each alignment groups. Alignment groups with a very poor maximum E-value ($> 0.001$) were discarded, as were groups overlapping with a much better alignments to a different isotype or exon type, where ``much better" was defined as a bitscore difference of at least 16.5. Where conflicting alignments to different isotypes or exon types co-occurred without a sufficiently large difference in bitscore, both alignment groups were retained for manual resolution of exon identity.

Following resolution of conflicts, alignment groups underwent a second filtering step of increased stringency, requiring a minimum E-value of $10^{-8}$ and at least two aligned reference exons over all reference species to be retained. Each surviving alignment group was then converted to a sequence range, extended by 10bp at each end to account for truncated alignments failing to cover the ends of the exon, and used to extract the corresponding exon sequence into FASTA format. These sequences then underwent manual curation to resolve conflicting exon identities, assign exon names and perform initial end refinement based on putative splice junctions.

In order to validate intron/exon boundaries and investigate splicing behaviour among \textit{IGH} constant-region exons, published RNA-sequencing data (\Cref{tab:rnaseq-sources}) was aligned to the annotated locus using STAR (\citep{dobin2013star}, version 2.5.2b). In both cases, reads files from multiple individuals were concatenated and aligned together, in order to make the intron/exon boundary changes in mapping behaviour as clear as possible. % TODO: Add more software versions, or a table at the end

\begin{table}
\caption{RNA-sequencing datasets used for \textit{IGH} constant-region exon refinement and isoform identification}
\centering
\begin{threeparttable}
\begin{tabular}{>{\bfseries}c|c|c}\toprule
Species & \textit{N. furzeri} & \textit{X. maculatus} \\\midrule
\multirow{2}{*}{Ages} & 6 weeks (4 individuals) & \multirow{2}{*}{Various\tnote{1}} \\
& 16 weeks (4 individuals) & \\\midrule
Tissues & Gut & Various\tnote{2}\\\midrule
BioProject Accession & PRJNA379208 & PRJNA420092\\\midrule
\multirow{26}{*}{SRA Run Accessions} & SRR5344350 & SRR6327069\\
& SRR5344343 & SRR6327070\\
& SRR5344344 & SRR6327071\\
& SRR5344345 & SRR6327072\\
& SRR5344346 & SRR6327073\\
& SRR5344347 & SRR6327074\\
& SRR5344348 & SRR6327075\\
& SRR5344349 & SRR6327076\\
& SRR5344350 & SRR6327077\\
&&SRR6327078\\
&&SRR6327079\\
&&SRR6327080\\
&&SRR6327081\\
&&SRR6327082\\
&&SRR6327083\\
&&SRR6327084\\
&&SRR6327085\\
&&SRR6327086\\
&&SRR6327087\\
&&SRR6327088\\
&&SRR6327089\\
&&SRR6327090\\
&&SRR6327091\\
&&SRR6327092\\
&&SRR6327093\\
&&SRR6327094\\\midrule
Source & \citep{smith2017microbiota} & Citation not given\\ % TODO: Contact authors re citation?
\bottomrule\end{tabular} % TODO: Table notes, sources
	\begin{tablenotes}
	\item[1] Ages used for \textit{X. maculatus} RNA-sequencing raised from embryonic (\dots) to \dots months; see source for details. % TODO: Specify source and details
	\item[2] Tissues used for \textit{X. maculatus} RNA-sequencing included brain, heart, liver, gut, skin or whole fish; see source for details. % TODO: Specify source
	\end{tablenotes}
\end{threeparttable}
\label{tab:rnaseq-sources}
\end{table}

Before aligning the RNA-seq reads, each locus underwent basic repeat masking, using the built-in zebrafish repeat parameters from the RepeatMasker program \citep{smith2016repeatmasker}:

\begin{lstlisting}[language=bash]
RepeatMasker -species danio -dir <masked_locus_dir> -s <unmasked_locus_path>
\end{lstlisting}

after masking, a STAR genome index was generated from each locus:

\begin{lstlisting}[language=bash]
STAR --runMode genomeGenerate --genomeDir <star_index_directory_path> --genomeFastaFiles <masked_locus_path> --genomeSAindexNbases <sa_index>
\end{lstlisting}

where the \lstinline{--genomeSAindexNbases} option determines the size of the suffix-array index and is dependent on the length of the reference sequence being indexed (\Cref{tab:star_sa_index}, \Cref{eq:sa_index}). 

\begin{equation}
\mathrm{SA~index~size~(bits)} = \frac{\log_2(\mathrm{length~of~reference~sequence})}{2} - 1
\label{eq:sa_index}
\end{equation}

\begin{table}
\centering
\caption{Reference sequences and corresponding SA index sizes used in RNA-seq alignment to \textit{IGH} loci}
\begin{tabular}{clc}\toprule 
Species & Reference & SA index size (bits)\\\midrule
\textit{X. maculatus} & IGH (whole locus) & 8\\
\textit{X. maculatus} & IGHZ constant region & 5\\ 
\textit{X. maculatus} & IGHM/D constant regions & 6\\ 
\textit{N. furzeri} & IGH (whole locus) & 8\\
\textit{N. furzeri} & IGHM/D constant regions & 6\\
\bottomrule
\end{tabular}
\label{tab:star_sa_index}
\end{table}

Following index generation, the RNA-seq reads were mapped to the generated index as follows:

\begin{lstlisting}[language=bash]
STAR --genomeDir <star_index_directory_path> --readFilesIn <input_reads> --outFilterMultimapNmax 5 --alignIntronMax 10000 --alignMatesGapMax 10000
\end{lstlisting}

where the \lstinline{--outFilterMultimapNmax} option excludes read pairs mapping to more than five distinct co-ordinates in the reference sequence and the \lstinline{--alignIntronMax} option excludes read pairs spanning predicted introns of more than 10kb, and the \lstinline{--alignMatesGapMax} option excludes read pairs mapping more than 10kb apart. Following alignment, the resulting SAM files were processed into sorted, indexed BAM files using SAMtools and visualised with Integrated Genome Viewer (IGV) to determine intron/exon boundaries of predicted exons, as well as the major splice isoforms present in each dataset.

In order to reduce time and memory requirements for generating alignment figures and reduce noise caused by reads mapping  across the constant regions, secondary alignments were performed on truncated loci consisting only of the IGHM/D or (where present) IGHZ constant regions, plus a few flanking kilobases on each side. In these cases, the additional parameters constraining multimapping, intron length and mate distance were not necessary due to the much shorter and less repetitive reference sequence.

\subsubsection{Downstream analysis}

% TODO: David's chromosomal synteny analysis here

Synteny between subloci in the \textit{N. furzeri} locus was analysed using \lstinline[language=R]{DECIPHER}'s standard synteny pipeline \citep{wright2016decipher}, which searches for chains of exact $k$-mer matches within two sequences:

\begin{lstlisting}[language=R]
DBPath <- tempfile()
DBConn <- dbConnect(SQLite(), DBPath)

Seqs2DB(seqs = <sublocus_1_sequence>, type = "XStringSet", dbFile = DBConn, identifier = "IGH1", verbose = FALSE)
Seqs2DB(seqs = <sublocus_2_sequence>, type = "XStringSet", dbFile = DBConn, identifier = "IGH2", verbose = FALSE)

dbDisconnect(DBConn)

SyntenyObject <- FindSynteny(dbFile = DBPath, verbose = FALSE)
\end{lstlisting}

Cross-locus sequence comparisons between gene segments in \textit{N. furzeri} were performed analogously to the comparisons involved in VH family assignment, with \lstinline[language=R]{pairwiseAlignment} and \lstinline[language=R]{pid} from \lstinline[language=R]{Biostrings} and the same pairwise-identity metric (\Cref{eq:pid}).

\subsection{Constant-region extraction and analysis in other cyprinodontiform species}
\label{sec:ch_char_crossspecies}

For species other than \textit{N. furzeri} or \textit{X. maculatus} (\Cref{tab:cyprinodontiform_genomes}), genome scaffolds identified as potentially containing \textit{IGH} gene segments (\Cref{sec:locus_scaffolds}) were directly subjected to constant-exon inference as described in the first two paragraphs of \Cref{sec:locus_char_ch}, using \textit{N. furzeri} and \textit{X. maculatus} as additional reference species. Intron/exon boundaries were predicted manually based on BLASTN and BLASTP alignments to closely-related species and the presence of conserved splice-site motifs (\texttt{AG} at the 5' end of the intron, \texttt{GT} at the 3' end \citep{shapiro1987splice}). In cases where no 3' splice site was expected to be present (e.g. for CM4 or TM2 exons), the nucleotide exon sequence was terminated at the first canonical polyadenylation site (\texttt{AATAAA} if present, otherwise one of \texttt{ATTAAA}, \texttt{AGTAAA} or \texttt{TATAAA} \citep{ulitsky2012polya}), while the amino-acid sequence was terminated at the first STOP codon. In many cases, it was not possible to locate a TM2 exon due to its very short (two-amino-acid-residue) conserved coding sequence. % TODO: Move last sentence to results

In addition to running this pipeline on the additional cyprinodontiform species, the same pipeline was run on the medaka and stickleback locus sequence isolated in \Cref{sec:ref_locus_ola,sec:ref_locus_gac}, in order to generate identically-processed CH databases from these reference species for phylogenetic reconstruction. This process successfully re-isolated all constant exons annotated in the published versions of the respective IGH loci, as well as several additional medaka exons that had either not been detected or excluded in the original study.

\subsection{Phylogenetic reconstruction}
\label{sec:phylo_methods}

\subsubsection{CH exon tree}
\label{sec:phylo_methods_ch}


To build a phylogenetic tree of CH exons, nucleotide and amino-acid constant-exon sequences from fourteen species (the twelve species specified in \Cref{tab:cyprinodontiform_genomes}, plus medaka and stickleback) were renamed to include the three-letter species code of the source species, then concatenated to generate a single sequence database for all species for each sequence type. As identical sequences can cause problems during phylogenetic analysis, exons with completely identical sequences (typically duplicates within a single locus) were collapsed together into a single FASTA sequence, which was relabelled with the names of all its parent sequences. Transmembrane exons and secretory tail annotations were discarded, as were sequences with more than 25\% missing characters; in addition, CM4 nucleotide sequences were trimmed to the coding region, removing the 3'-UTR. Remaining sequences were then aligned with \lstinline{PRANK}:
% TODO: rework genome table to include species codes and reference species

\begin{lstlisting}
prank -d=<ch_fasta> -o=<output_prefix> -F -termgap
\end{lstlisting}

where \lstinline{-F} determines how the program handles predicted indel mutations and \lstinline{-termgap} specifies that terminal gaps should be penalised in the same way as internal ones. The resulting alignments were then passed to the maximum-likelihood phylogenetic inference program \lstinline{RAxML} (\citep{stamatakis2005raxml3,stamatakis2006raxml6,stamatakis2014raxml8}, version 8.2.12), using the SSE3-enabled parallelised version of the software and ..:

\begin{lstlisting}
raxmlHPC-PTHREADS-SSE3 -f a -m <model> -s <ch_prank_alignment> -w <output_dir> -N 100 -x <bootstrap_seed> -p <parsimony_seed> -n <sequence_type>
\end{lstlisting}

where \lstinline{-f a} specifies a single-run ML analysis with rapid bootstrapping, \lstinline{-N 100} specifies that 100 bootstrap replicates should be performed, \lstinline{-p} and \lstinline{-x} specify the random seeds for the initial parsimony tree and rapid-bootstrapping process, respectively, and  \lstinline{-n} specifies the output file suffix (here indicating the sequence type, either \lstinline{nt} or \lstinline{aa}). For nucleotide alignments, the standard \lstinline{GTRGAMMA} model was used, while for protein alignments the flexible \lstinline{PROTGAMMAAUTO} model was used, allowing the program to select the substitution model which best fit the initial parsimony tree. After tree inference, \dots % TODO: Tree visualisation and analysis methods here

\subsubsection{VH and JH trees}

To investigate the evolutionary relationships between VH segments (and particularly VH families) in \textit{N. furzeri} and \textit{X. maculatus}, segments from these two species underwent phylogenetic analysis in a similar manner to the CH exons in \Cref{sec:phylo_methods_ch}. Segments from both species were relabelled to include the appropriate three-letter species code, concatenated together, processed to remove identical sequences, and filtered to remove sequences with more than 25\% missing characters. The remaining sequences were aligned together with \lstinline{PRANK}, and the resulting alignment was used to perform phylogenetic analysis with RAxML, in both cases using the same parameters specified in \Cref{sec:phylo_methods_ch}. An identical analysis was performed for JH segments from both species, for a total of four independent trees (VH-NT, VH-AA, JH-NT and JH-AA). \dots % TODO: Tree visualisation and analysis methods here

\subsubsection{Species-tree generation and annotation}

\dots

\subsection{Data visualisation}

Unless otherwise specified, data were visualised using \lstinline[language=R]{ggplot2} \citep{wickham2016ggplot2}. Chromosome ideograms, locus structure visualisations, and sashimi plots were constructed using \lstinline[language=R]{Gviz} \citep{hahne2016gviz}. Cluster dendrograms and phylogenetic trees were drawn with \lstinline[language=R]{ggtree} \citep{guangchuang2018ggtree}, using utilities from \lstinline[language=R]{ape} \citep{paradis2018ape} and \lstinline[language=R]{tidytree} \citep{guangchuang2018tidytree}. Sequence logos were drawn with \lstinline[language=R]{ggseqlogo} \citep{wagih2017ggseqlogo}. 

\section{The \textit{IgH} locus in \textit{Nothobranchius furzeri}}
\label{sec:nfu-locus}


	\subsection{Overall structure}
	\label{sec:nfu-locus-structure}
	
	The turquoise killifish genome contains a single \textit{IgH} locus approximately 305 kilobases in length, located towards the 3' end of chromosome 6 (\Cref{fig:nfu-locus-map-a}).
	This locus comprises two complete subloci, IGH1 (155 kb) and IGH2 (118 kb), present in tandem and each occupying a classic ${VH-DH-JH-CH}$ translocon configuration. This modified translocon structure, with multiple translocon subloci present in tandem, has been observed in a number of teleost \textit{IgH} loci including catfish, medaka and stickleback. Unusually, however, the smaller IGH2 sublocus in TK \textit{IgH} is present in antisense relative to the larger IGH1, with the two subloci beginning at opposite ends of the locus and facing each other in the middle (\Cref{fig:nfu-locus-map-b}). Such a multi-orientation locus structure has only previously been observed in medaka, and there only in a small, partially-assembled sublocus of uncertain functionality; it is interesting to see this unusual feature reproduced here, in a fairly closely-related species, and it is possible that this ideosyncracy is homologous between the two loci. % TODO: Use RNA-seq data to check functionality of IGH2
	
	Compared to other closely-related loci, the killifish locus is relatively sparse and simple, with comparatively low functional complexity relative to its overall size. For example, whereas the stickleback locus fits four subloci, 49 V segments and 10 constant regions into c. 200 kb, the killifish locus, despite being ~50\% longer, contains only 2 subloci, four constant regions and 24 V segments (including pseudogenised Vs). This difference results from the unusually large amount of nonfunctional sequence padding the killifish locus, resulting in large gaps between variable segments and in some cases between constant-region exons (\Cref{fig:nfu-locus-map-b}); this high prevalence of repetitive DNA is consistent with the rest of the TK genome, which comprises more than 60\% repetitive sequence, compared to just over 15\% in stickleback.
	
	The two subloci in the turquoise killifish locus are generally highly similar in their functional sequence, with a high degree of synteny between their functional regions (\Cref{fig:nfu-locus-synteny}). The greatest degree of divergence occurs in the VH and DH regions, with what appear to be repeated deletion events in IGH2 resulting in a substantially lower number of VH and DH segments compared to IGH1; conversely, the JH and constant regions are almost identical between two subloci. These patterns are discussed in more detail in \Cref{sec:nfu-locus-constant,sec:nfu-locus-variable}.
	
	% TODO: Can I say anything about the history of the locus? Ask David.
	
	\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{/home/will/Documents/code/figures/2018-11-06/nfu-locus-map_small} % TODO: Move to thesis directory, switch to large figure
			    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:nfu-locus-map-a}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:nfu-locus-map-b}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:nfu-locus-map-c}
        \end{subfigure}
	\caption[The immunoglobulin heavy chain (\textit{IGH}) locus in \textit{Nothobranchius furzeri}]{\textbf{The immunoglobulin heavy chain (\textit{IgH}) locus in \textit{Nothobranchius furzeri}:} (A) Position of the \textit{IgH} locus on chromosome 6 of the \textit{N. furzeri} genome. (B) Arrangement of VH, DH, JH and constant-region gene segments on the \textit{N. furzeri} \textit{IgH} locus. All segments follow the orientation of their parent sublocus, indicated in the uppermost track. (C) Detailed map of the constant regions of the \textit{IGH1} sublocus, indicating the position and identity of the constant-region exons and the exon composition of expressed \textit{IgH} isoforms in the turquoise killifish.}
	\label{fig:nfu-locus-map}
	\end{figure}
	
	\begin{figure}
	\centering
	\includegraphics[width=0.9\textwidth]{_Figures/png/nfu-locus-dots.png}
	\caption[Sequence homology between subloci in \textit{N. furzeri} \textit{IGH}]{\textbf{Sequence homology between subloci in \textit{N. furzeri} \textit{IGH}:} Synteny plot of sequential best matches between \textit{IGH1} and \textit{IGH2} subloci, with gene segment regions indicated by coloured rectangles along each axis.}
	\label{fig:nfu-locus-synteny}
	\end{figure}
	
	\subsection{Constant regions}
	\label{sec:nfu-locus-constant}
	
	The isotype (also known as the class) of an antibody determines its functional role within the immune system, including its possible effector functions and whether it can be secreted. Three antibody isotypes have been identified to date in teleost fishes: IgM, IgD and IgZ (a.k.a. IgT, IgT/Z or IgZ/T). Of these, IgM and IgD are highly primitive within the jawed vertebrates and found in most or all other vertebrate groups; within the teleosts, both appear to be universal. Conversely, IgZ is a teleost-specific isotype which is absent in other vertebrate taxa; within the teleosts, most characterised exons possess IgZ, but at least two (medaka and channel catfish) have been found to lack it. In rainbow trout, IgZ has been found to play a specialised mucosal role in the immune system analagous to that of IgA in mammals, and it is widely assumed to play this specialised role throughout the teleosts; it is as yet unclear how mucosal immunity is effected in species lacking IgZ.
	
	IgM, the most primitive and widely-found isotype in jawed vertebrates, is present immediately downstream of the main JH-region and occupies the standard six-exon configuration, with four \cm{} exons and two transmembrane domains present in series on the chromosome (\Cref{fig:nfu-locus-map-b,fig:nfu-locus-map-c,fig:teleost-igm-exons-a}). As with other species, both secreted and transmembrane isoforms of IgM are present in the transcriptome, with secreted IgM consisting of \cm{1-4} (\Cref{fig:nfu-locus-map-c,fig:nfu-locus-sashimi-a,fig:teleost-igm-exons-b}); however, the exon configuration of transmembrane IgM deviates from both that seen in mammals (in which exon TM1 is spliced to a cryptic splice site within \cm{4}) and most teleosts (in which the canonical splice site following \cm{3} is used and \cm{4} is excised). Rather, TK IgM-TM resembles that of medaka, in which both \cm{3} and \cm{4} are excluded and the canonical splice site at the end of \cm{2} is spliced directly to TM1 (\Cref{fig:teleost-igm-exons-c,fig:teleost-igm-exons-d,fig:teleost-igm-exons-e}). The similarity to medaka, which again is the closest relative of TK with a characterised locus, raises the possibility that this unusual splicing behaviour may be a conserved feature of both lineages; however, the underlying mechanism giving rise to this difference in splicing behaviour is unknown.
	
		Unlike IgM, the exon structure of IgD is highly variable across the teleosts, ranging from roughly 7-17 \cd{} exons in addition to the transmembrane domains. The core structure of IgD comprises seven \cd{} exons (\cd{1-7}), but some subset of these exons may be missing or duplicated in any given species -- in medaka, for example, \cd{5} is missing in all subloci, while in many species (e.g. zebrafish, salmon, and channel catfish) \cd{2-4} are duplicated in two or more tandem blocks. This latter configuration is a in turquoise killifish, in which the IgD constant region immediately follows IgM in both subloci and has a 

\cd{1}-(\cd{2}-\cd{3}-\cd{4})$_2$-\cd{5}-\cd{6}-\cd{7}-TM1-TM2 

	\noindent configuration, for a total of 12 exons per IgD constant region (\Cref{fig:nfu-locus-map-b,fig:nfu-locus-map-c}). All of these exons appear to be expressed in tandem, resulting in a much longer IgH transcript than is observed for any isoform of IgM (\Cref{fig:nfu-locus-map-c,fig:nfu-locus-sashimi-b}). Of the duplicated exon pairs, \cm{3}A/B and to a lesser extent \cm{4}A/B are highly similar, while \cm{2}A/B are more distinct (\Cref{fig:nfu-ch-aln}). As in other teleost species, IgD in the turquoise killifish includes a chimeric \cm{1} exon at the 5' end of the constant-region transcript, for a total of 13 exons per IgD-TM mRNA (\Cref{fig:nfu-locus-sashimi-b}).

	While the best-known form of IgD in teleosts is transmembrane, secreted IgD has been observed in at least two teleost species, with different mechanisms used in each case: in channel catfish, one dedicated sublocus has a dedicated IgD secretory exon in place of the transmembrane exons, while in rainbow trout (and possibly some other species like Atlantic salmon and cod) a run-on event at the end of \cd{7} results in the production of a secretory tail in a manner similar to secretory IgZ. However, neither a specialised secretory exon nor a \cd{7} secretory tail could be detected in turquoise killifish, suggesting that IgD may only be expressed in transmembrane form in this species. % TODO: Figure comparing post \cd{7} sequence in trout, salmon, cod and TK (and other species from paper?) to demonstrate lack of secretory tail in TK
	
	In the case of both IgM and IgD, the constant regions are present in their expected configuration in each sublocus and are highly similar in sequence between the subloci, with an average of 98.4\% nucleotide sequence identity for corresponding IgM exons and 99.3\% for corresponding IgD exons (\Cref{fig:nfu-ch-aln} and \Cref{tab:nfu-ch-aln}).

	The most striking feature of the constant-region structure of the \textit{IGH} locus in \textit{N. furzeri} is the complete absence in both subloci of IgZ. As discussed above, IgZ constant exons have been found to be present in the great majority of teleost \textit{IgH} loci characterised to date, with the only previous exceptions being channel catfish and medaka. Its absence in turquoise killifish (\Cref{fig:nfu-locus-map-b}) is therefore notable, and immediately raises questions about the nature, kinetics and efficacy of mucosal adaptive immunity in this species. Since medaka, the closest relative of TK with a characterised locus, also lacks IgZ, its absence in TK also raises the question of whether its absence occurred in a common ancestor to both species or is the result of independent deletion events in both lineages; this question is addressed in more depth in subsequent sections of this chapter. % IgZ paragraph at start or end of subsection?	
		

	% TODO: Table and figure of antibody isoforms in TK: length, molecular weight, number of C-regions, etc. (in appendix)
	
	\begin{figure}
	\centering
		    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:teleost-igm-exons-a}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:teleost-igm-exons-b}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:teleost-igm-exons-c}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:teleost-igm-exons-d}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:teleost-igm-exons-e}
    \end{subfigure}
	\includegraphics[width=0.8\textwidth]{_Figures/png_edited/teleost-igm-exons}
	\caption[IgM exon usage in other vertebrates]{\textbf{IgM exon usage in other vertebrates:} Schematic of IgM splice patterns in different isoforms and taxonomic groups; (A) standard genomic (pre-splicing) configuration of IgM, following VDJ recombination; (B) exon configuration of secreted IgM (IgM-S) in tetrapods and teleosts; (C) exon configuration of transmembrane IgM (IgM-TM) in tetrapods, demonstrating the use of a cryptic splice site in \cm{4}; (D) standard IgM-TM exon configuration in teleosts, demonstrating the direct splicing of \cm{3} to TM1 and exclusion of \cm{4}; (E) unusual IgM-TM exon configuration observed in medaka, in which both \cm{3} and \cm{4} are excluded. Figure adapted from Fillatreau \textit{et al.} (2013).}
	\label{fig:teleost-igm-exons}
	\end{figure}
	
	\begin{figure}
	\centering
		\begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:nfu-locus-sashimi-a}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:nfu-locus-sashimi-b}
    \end{subfigure}
	\includegraphics[width=\textwidth]{_Figures/png/nfu-locus-sashimi}
	\caption[Constant-region isoforms in \textit{N. furzeri}]{\textbf{Constant-region isoforms in \textit{N. furzeri}:} Coverage and sashimi plots of STAR-aligned RNA-seq reads from \textit{N. furzeri} gut samples, demonstrating the splicing behaviour of IGH1 constant-region isoforms. (A) IGHM exon splicing, showing alternative splicing patterns of IGHM-TM and IGHM-S; (B) IGHD exon splicing, including splicing of \cm{1} to \cd{1}.}
	\label{fig:nfu-locus-sashimi}
	\end{figure}
	
	\begin{figure}
	\centering
	\includegraphics[width = \textwidth]{_Figures/png/nfu-ch-aln}
	\caption[Cross-sublocus sequence similarity in constant-region exons in \textit{N. furzeri} \textit{IGH}]{\textbf{Cross-sublocus sequence similarity in constant-region exons in \textit{N. furzeri}:} Heatmap of percentage sequence identity between amino-acid (right) and nucleotide (left) sequences of constant-region exons (excluding IGHM-TM2 and IGHD-TM2) from the two subloci of \textit{N. furzeri} \textit{IGH}, calculated using pairwise Needleman-Wunsch global alignments. Note the very high similarity between corresponding exons of the two subloci, as well as the substantial similarity between IGHD-3A/3B and IGHD-4A/4B in both subloci.}
	\label{fig:nfu-ch-aln}
	\end{figure}
	
	\begin{table}\centering
		\caption[Cross-sublocus sequence similarity between corresponding constant-region exons in \textit{N. furzeri} \textit{IGH}]{\textbf{Cross-sublocus sequence similarity in constant-region exons in \textit{N. furzeri}:} Percentage sequence identities of pairwise Needleman-Wunsch global alignments between nucleotide (NT) or amino-acid (AA) sequences of corresponding exons from the two subloci of \textit{N. furzeri} \textit{IGH}.}
	\input{_Figures/tables/nfu-ch-aln}
	\label{tab:nfu-ch-aln}
	\end{table}
	




	\subsection{Variable regions}
	\label{sec:nfu-locus-variable}
	
	\subsubsection{Organisation, functionality and diversity}
	\label{sec:nfu-locus-variable-orga}
	 
	% Presence and functionality
	In total, 24 VH-segments, 14 DH-segments and 17 JH-segments were identified in the \textit{N. furzeri} locus, of which the majority (17 VH, 10 DH and 8 JH) were present in \textit{IGH1}. Of the VH segments identified, three contain premature STOP codons, though none is out-of-frame; conversely, all the DH and JH segments identified appear to be in-frame and functional, with no premature STOP codons. However, in all cases a minority of segments contain RSS sequences that deviate significantly from the expected consensus sequence; it is unclear whether these sequences can recombine to successfully produce mature VDJ sequences \textit{in vivo}. In the case of the VH segments, of the six sequences without clearly functional RSS sequences, three also contain premature STOP codons, suggesting the changes to the RSS in these cases may arise from relaxed purifying selection on already-pseudogenised sequences.
	
	% Location and exceptions
	Of the VH, DH and JH segments identified, all but one of each type of segment is located within contiguous V-, D-, and J-regions within each sublocus, supporting a modified translocon configuration for killifish \textit{IGH}. The exceptions to this are IGH1D01 and IGH1J01, which are embedded within the IGH1 V-region, and a single VH segment located in between the IGHD contant regions of the two subloci (\Cref{fig:nfu-locus-map-b}). The unusual location of IGH1D01 and IGH1J01 may represent the result of a transposition event within the IGH locus; however, their close colocalisation and 5' position within the IGH1 sublocus, as well as the fact that neither has a close paralogue in IGH2 (\Cref{fig:nfu-dj-alignment-b}), suggest that they may alternatively represent the remnant of a formerly present IGHZ constant region, as these typically have dedicated D/J segments independent of those serving IGHM. Given its forward orientation, meanwhile, the orphaned VH-segment was assigned to the IGH1 sublocus as IGH1V1-07; however, if annotated correctly, it is unlikely to successfully recombine with segments in either sublocus due to its unusual location. % Alternatively, the unusual location of this VH segment could be due to either a misassembly or the presence of an additional sublocus missing from the current assembly; however, given the structurally volatile nature of the IGH locus and the frequency with which isolated VH loci are found in other parts of the genome, the presence of an orphaned locus in between two functional subloci is not implausible.
		
	% Segment numbers
	The total number of functional VH segments in the killifish locus is unusually small in comparison to the total numbers observed in many other teleost species (\Cref{tab:teleost-vh-counts}); however, the number of segments per sublocus is in line with the numbers seen in closely-related species (2 to 12 in medaka, 6 to 18 in stickleback), with the overall difference mainly arising from a difference in the number of subloci per locus. A similar pattern is observed with DH and JH segments, with similar numbers of segments per sublocus in killifish and closely-related species, especially medaka. It therefore appears that the per-sublocus segment diversity available to the turquoise killifish is similar to that of previously characterised species, with any difference in total available diversity at this level arising from differences in the number of functional subloci rather than the size of the V/D/J-regions \textit{per se}.
	
	\begin{table}
	\begin{threeparttable}
	\caption{\textbf{Number of functional VH-segments and VH-families in other teleost species}}
	\label{tab:teleost-vh-counts}
	\begin{tabular}{ccccc}\toprule
	\textbf{Common Name} & \textbf{Species} & \textbf{\# Functional VH Segments} & \textbf{\# VH Families} & \textbf{Source}\\\midrule
	Zebrafish & \textit{Danio rerio} & 39 & 13\,\tnote{1} & Magadan 2015 \\
	Grasscarp & \textit{Ctenopharyngodon idella} & 8 & 5\,\tnote{2} & Xiao 2010 \\
	Fugu & \textit{Takifugu rubripes} & 34 & 3 & Magadan 2015 \\
	Medaka & \textit{Oryzias latipes} & 35 & 6 & Fillatreau, Magadan 2011 \\
	Stickleback & \textit{Gasterosteus aculeatus} & 49 & 4 & Magadan 2015 \\
	Turquoise killifish & \textit{Nothobranchius furzeri} & 21\,\tnote{3} & 6 & -- \\
	\bottomrule\end{tabular}
	\begin{tablenotes}
	\item[1] VH families in zebrafish identified based on 70\% (rather than 80\%) sequence identity.
	\item[2] It is not clear what clustering method or threshold was used to identify VH families in grasscarp.
	\item[3] Excluding VH segments with nonsense or frameshift mutations, but not those with uncertain or missing RSS sequences.
	\end{tablenotes}
	\end{threeparttable}
	\end{table} % TODO: Convert source labels into proper citations
	
	\subsubsection{Sublocus synteny and evolution}
	\label{sec:nfu-locus-variable-synteny}

	% Synteny
		As can be seen from \Cref{fig:nfu-locus-synteny}, much of the V-, D- and especially J-region sequence in the \textit{N. furzeri} locus is syntenic between the two \textit{IGH} subloci, with downstream portions of the \textit{IGH2} V-region corresponding to downstream parts of the \textit{IGH1} region (after correcting for locus orientation). Of the seven VH segments in \textit{IGH2}, six have a correspending segment on \textit{IGH1} with which they share at least 97\% sequence identity (\Cref{fig:nfu-vh-families}), and these partner segments are largely (but not entirely) colinear in their ordering between the two subloci. A similar pattern can be observed for the D- and (especially) the J-regions: of the four DH segments detectable in IGH2, three (IGH2D02 to IGH2D04) are identical with another block of adjacent DH segments in IGH1 (IGH1D05 to IGH1D07), while the the J-regions exhibit almost complete sequence identity between the eight JH segments of the main JH region in IGH1 and the eight JH segments in IGH2 (\Cref{fig:nfu-dj-alignment}).
		
		Nevertheless, as is clear from \Cref{fig:nfu-locus-synteny}, there are large portions the \textit{IGH1} variable region, including the first 25 kilobases of the V-region, for which no corresponding sequence exists in \textit{IGH2}, and there are many VH and DH segments in IGH1 (and a much smaller number in IGH2) for which no close homologue exists in the other sublocus. Taken together, these data are consistent with a model in which \textit{IGH2} was produced via duplication and inversion of all or part of \textit{IGH1}, followed by subsequent deletion events in the redundant, and structurally volatile, \textit{IGH2} VH and DH regions. However, it is not clear at present how to distinguish between this model and an alternative one of expansion in IGH1, explain why IGH2 and IGH1 should have undergone deletion events, or identify why the JH region is so much more conserved between subloci than either the VH or JH regions.
% TODO: Think about these questions, try to suggest hypotheses
		

	\subsubsection{VH families}
	\label{sec:nfu-locus-variable-families}
	
	VH sequences within an \textit{IGH} locus are conventionally grouped into families on the basis of nucleotide sequence identity, with a typical identity cutoff of 80\%. In order to group the \textit{N. furzeri} VH genes into families, pairwise Needleman-Wunsch global alignments were performed on each pair of VH sequences to obtain pairwise identity scores, followed by single-linkage clustering on the resulting identity matrix. Cutting the dendrogram at 80\% sequence identity revealed a total of six VH families, of which four contained more than one VH segment (\Cref{fig:nfu-vh-families}). 	V2 and V4 are highly similar and all the members of V4 are pseudogenised by premature STOP codons; it may therefore be better to regard V4 as a pseudogenised subfamily of V2 than a VH family in its own right. Of the remaining families, V1 and V2 make up the bulk (42\% and 29\% respectively) of the VH segments in the locus. 
	
	The number of VH families is roughly in line with those found in related species (\Cref{tab:teleost-vh-counts}), indicating that, while the total number of VH segments in the turquoise killifish may be unusually low (see \Cref{sec:nfu-locus-variable-orga}), the degree of sequence diversity found \textit{among} these segments is in line with other, previously-characterised teleost species. 
	
	\begin{figure}
	\centering
	\begin{subfigure}{0em}
	\phantomsubcaption{}
    \label{fig:nfu-vh-families-a}
    \end{subfigure}
    \begin{subfigure}{0em}
    \phantomsubcaption{}
    \label{fig:nfu-vh-families-b}
    \end{subfigure}
	\includegraphics[width=0.8\textwidth]{_Figures/png/nfu-vh-families.png}
	\caption[VH families in the in \textit{N. furzeri} \textit{IgH} locus]{\textbf{VH families in the in \textit{N. furzeri} (\textit{IgH}) locus:} (A) Dendrogram of sequence similarity of VH segments in the \textit{N. furzeri} locus, arranged by single-linkage clustering on nucleotide sequence identity. The red line indicates the 80\% cutoff point for family assignment, while branch colour indicates family membership. (B) Heatmap of family relationships among \textit{N. furzeri} VH segments, with coloured shading indicating families and red dots indicating pairwise nucleotide sequence identity of at least 80\%. In both subfigures, VH families containing multiple segments are coloured, single-segment families are in grey, and segments from the IGH2 sublocus are displayed in \textbf{boldface}.}
	\label{fig:nfu-vh-families}
	\end{figure}

	% TODO: Detailed V-co-ordinate table (in appendix)	

\begin{figure}
\centering
	\centering
	\begin{subfigure}{0em}
	\phantomsubcaption{}
    \label{fig:nfu-dj-alignment-a}
    \end{subfigure}
    \begin{subfigure}{0em}
    \phantomsubcaption{}
    \label{fig:nfu-dj-alignment-b}
    \end{subfigure}

\includegraphics[width=0.6\textwidth]{_Figures/png/nfu-dj-aln}
\caption[Cross-sublocus sequence similarity in DH and JH gene segments in \textit{N. furzeri}]{\textbf{Cross-sublocus sequence similarity in DH and JH gene segments in \textit{N. furzeri}:} Heatmap of percentage nucleotide sequence identities of Needleman-Wunsch global alignments between (A) DH and (B) JH gene segments in IGH1 vs IGH2, revealing syntenic runs of highly similar sequences across both subloci.}
\label{fig:nfu-dj-alignment}
\end{figure}

	\subsubsection{Recombination signal sequences}
	
	In their heptamer sequences, nonamer sequences, and spacer lengths, the recombination signal sequences (RSS) marking the ends of the VH, DH and JH gene segments in the \textit{N. furzeri} locus strongly resemble those of other characterised teleosts, which in turn match those of non-teleost loci (\Cref{fig:nfu-rss-seqlogo-all}). The overall heptamer and nonamer consensus sequences (\texttt{CACAGTG} for heptamers and \texttt{ACAAAAACC} for nonamers) closely matched those expected from the literature, while in 88\% of cases the spacer region was within 1bp of the expected length (12bp for D-RSSs, 23bp for V- and J-RSSs); unexpectedly, the greatest number of VH-RSSs had a 22bp (rather than 23bp) spacer, but this is not expected to interfere with RSS functionality. Overall, the RSSs in the turquoise killifish appear to be supporting the normal operation of VDJ-recombination in this species.
	
	\begin{figure}
	\includegraphics[width=\textwidth]{_Figures/png/nfu-rss-seqlogo-all}
	\caption[Recombination signal sequences in the \textit{N. furzeri} \textit{IGH} locus]{\textbf{Recombination signal sequences in the \textit{N. furzeri} \textit{IGH} locus:} (A) Sequence composition of conserved heptamer sequences across all \textit{N. furzeri} heavy-chain RSSs; (B) length distribution of unconserved spacer sequences in \textit{N. furzeri} heavy-chain RSSs; (C) sequence composition of conserved heptamer sequences across all \textit{N. furzeri} heavy-chain RSSs.}
	\label{fig:nfu-rss-seqlogo-all}
	\end{figure} % TODO: Add in row showing expected consensus sequence with highlighted conserved positions
	
	
	% TODO: Killifish V/D/J RSS sequence logos (separately, in appendix)



	
	\subsection{Discussion}
	
	The turquoise killifish IGH locus shares many features with other characterised teleost loci, including a modified tandem-translocon configuration with intact V, D, J and constant regions (\Cref{fig:nfu-locus-map-b}), four-exon secreted configuration of immunoglobulin M (\Cref{fig:nfu-locus-sashimi-a}), expanded IgD constant region with tandem \cd{}-exon block repeats (\Cref{fig:nfu-locus-sashimi-b,fig:nfu-locus-map-b},), a conserved RSS structure (\Cref{fig:nfu-rss-seqlogo-all}), and a chimeric \cm{1} in IGHD \Cref{fig:nfu-locus-sashimi-b}). However, it also exhibits many ideosyncratic features that differ from those observed in most characterised teleost loci, including an unusually small number of VH segments (\Cref{fig:nfu-vh-families} and \Cref{tab:teleost-vh-counts}), a four-exon \cm{1}-\cm{2}-TM1-TM2 configuration of transmembrane IgM (\Cref{fig:nfu-locus-sashimi-a}), an inverted sublocus present in antisense (\Cref{fig:nfu-locus-map-b}), and a complete absence of immunoglobulin Z.
	
	Many of these peculiarities, including the unusual IgM-TM splicing pattern, inverted sublocus, and lack of IgZ, are shared with the sublocus of medaka (\textit{Oryzias latipes}), which is the closest relative of \textit{Nothobranchius furzeri} to have its immunoglobulin heavy chain locus characterised prior to this work. Given this close relationship between turquoise killifish and medaka, an obvious hypothesis for the evolutionary history of these unusual traits was of a common origin in the common ancestor of the Cyprinodontiformes (the clade that contains the killifishes) and the Beloniformes (the clade that contains medaka). % Tree for this 

	To investigate this hypothesis further, I performed a complete characterisation of the \textit{IgH} locus in the platyfish \textit{Xiphophorus maculatus} (\Cref{sec:xma-locus}), another cyprinodontiform species that has seen widespread use as a model organism. Alongside this, I performed a more limited characterisation of the \textit{IgH} constant regions in 9 %?
	additional cyprinodontiform and beloniform species, to assess the broader patterns of immunoglobulin heavy chain evolution within these groups (\Cref{sec:comparative-loci}). The results of these additional investigations repeatedly called into question the hypothesis of homology in IgH locus ideosyncracies between medaka and turquoise killifish, and suggest a model of rapid locus evolution within this diverse and important teleost clade.
	
	% TODO: Relationship between killifish Vs and other species
	% TODO: Evolution of IGH1 vs 2
	% TODO: Discussion of mucosal immunity in absence of IGZ
		% TODO: Discuss smallness of killifish locus in the context of relaxed purifying selection and short life span
	% TODO: Section on "Investigating the evolutionary relationship between IGH1 and IGH2"
	% Keep for discussion:
%\q{Given the lack of IgZ in this species, how do you think they carry out mucosal immunity?}
%
%I'd assume using IgM. It's the primitive antibody and is expressed in secreted form in the killifish gut. I haven't investigated the protein expression so it's hard to say for sure, but I'd guess that's the answer. It's also quite possible, lacking a specialised mucosal class, that the answer is ``not very well". It would be very interesting to compare mucosal immune function in species with and without IgZ, especially if those species are closely related.
%
%Given the impressive results indicating the specificity of IgT for mucosal immunity in trout, and the lack of mucosal response of IgM in that species, it would be very interesting to see what mucosal responses look like in a species that lacks that isotype -- is IgM much more responsive and expressed in the gut than in IgT-possessing species, or is the mucosal response just worse?
	% (Incidentally, it would be great to see whether any B-cells in TK show recombinations in both subloci, vs one or the other. Single-cell DNA sequencing could plausibly do this.)



\section{The \textit{IgH} locus in \textit{Xiphophorus maculatus}}
\label{sec:xma-locus}

	\subsection{Overall structure}
	
	The \textit{X. maculatus} \textit{IGH} locus occupies roughly 263 kb on chromosome 16 (\Cref{fig:xma-locus-map-a}), in a region syntenic with the part of \textit{Nothobranchius furzeri} chromosome 6 containing the turquoise-killifish IGH locus. % TODO: Get a synteny plot of this, based on David's data.
	Unlike in \textit{N. furzeri}, and medaka, only a single ``sublocus" is present, which spans the entire length of the locus region; no evidence of a second, antisense locus could be identified. The overall structure of this locus roughly corresponds to that of simpler that of simpler teleost loci like those of zebrafish and fugu, albeit with some striking ideosyncracies, and it differs from that of \textit{N. furzeri} in several important respects.
	
	\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{/home/will/Documents/code/figures/2018-11-07/xma-locus-map}
	\caption[The immunoglobulin heavy chain (\textit{IGH}) locus in \textit{}]{\textbf{The immunoglobulin heavy chain (\textit{IGH}) locus in \textit{Xiphophorus maculatus}:} (A) Position of the \textit{IGH} locus on chromosome (group) 16 of the \textit{X. maculatus} genome. (B) Arrangement of VH, DH, JH and constant-region gene segments on the \textit{X. maculatus} \textit{IgH} locus. (C) Detailed map of the \textit{IGHZ} and \textit{IGHM/D} constant regions, indicating the position and identity of the constant-region exons and the exon composition of expressed \textit{IgH} isoforms in \textit{X. maculatus}.}
	\begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-locus-map-a}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-locus-map-b}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-locus-map-c}
    \end{subfigure}
	\label{fig:xma-locus-map}
\end{figure}

	
	\subsection{Constant regions}
	
	Most strikingly, unlike the turquoise killifish, the platyfish locus contains a fully intact IGHZ constant region, with a six-exon structure homologous with that seen in most other teleost loci (unlike in stickleback, the exon corresponding to \cz{2} is not missing) and a secretory tail produced by a run-on event from \cm{4} (\Cref{fig:xma-locus-map-b,fig:xma-locus-map-c}). Conversely, the genomic configurations of the IGHM and IGHD constant regions were highly similar to those of turquoise killifish, with a {\cm{1}-\cm{2}-\cm{3}-\cm{4}-TM1-TM2} configuration for IGHM and a {\cd{1}-(\cd{2}-\cd{3}-\cd{4})$_2$-\cd{5}-\cd{6}-\cd{7}-TM1-TM2} configuration for IGHD. The evolution of constant-region exon structure in the cyprinodontiforms, including the pattern of (\cd{2}-\cd{3}-\cd{4}) duplications, is discussed more in \dots .
	
	In order to investigate the expressed isoforms present in \textit{X. maculatus}, published RNA-sequencing reads from platyfish \dots % Tissues used, reference
	were aligned to the IGHZ and IGHM/D constant regions with STAR. % TODO: Refer to methods here
	The results indicate the expected splice patterns for IGHZ and IGHD, including a six-exon configuration for IGHZ-TM (\Cref{fig:xma-locus-sashimi-z}), the use of a \bp{23} secretory tail produced by a run-on event from \cz{4} in IGHZ-S, and the presence of a chimeric \cm{1} exon in IGHD-TM (\Cref{fig:xma-locus-sashimi-d}). However, while secretory IGHM (IGHM-S) in platyfish was found to occupy the expected four-exon configuration seen in turquoise killifish and elsewhere (\Cref{fig:xma-locus-sashimi-m}), the configuration observed for transmembrane IGHM (IGHM-TM) did not correspond to the four-exon structure shared between turquoise killifish and medaka (\Cref{fig:nfu-locus-sashimi-a}). Rather, IGHM-TM in \textit{X. maculatus} occupies the standard five-exon configuration seen in most characterised teleosts (\Cref{fig:teleost-igm-exons-d}). This surprising difference raises not only the question of the evolutionary history of this ideosyncratic splicing configuration, but also that of what, if any, functional difference arises from the presence or absence of \cm{3} in transmembrane IgM in different species. % TODO: Discuss possible experiments?

	The examination of the constant-region structures present in the \textit{X. maculatus} IGH locus therefore reveals three notable features that differ in state between medaka, turquoise killifish, and the southern platyfish. Most strikingly, the absence of IGHM in medaka and killifish, but its presence in platyfish, strongly suggests that the isoform has been lost independently in the two groups. Conversely, the presence of a four-exon IGHM-TM in medaka and killifish, but a five-exon configuration in platyfish, is less clear-cut: it may indicate an independent change in medaka and killifish that is absent in platyfish, but may also plausibly represent a reversion in platyfish to the more-primitive five-exon state. Without a deeper understanding of the currently-unknown physical underpinnings of this observed difference in splicing behaviour, this question can only be addressed through analysis of RNA-seq data from a large number of related species, something beyond the scope of the present investigation. % TODO: Do it anyway?
	Finally, the presence of a (\cd{2}-\cd{3}-\cd{4}) duplication in both platyfish and killifish, but not in medaka, suggests that this may be a shared primitive feature of the Cyprinodontiforms; however, given the apparent recurrence of this duplication pattern in many different groups across the teleosts, a strong conclusion cannot be drawn without a higher-resolution phylogenetic analysis of a larger number of related species. % TODO: See section ... etc.
	
			\begin{figure}
	\includegraphics[width=\textwidth]{_Figures/png/xma-locus-sashimi}
	\begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-locus-sashimi-m}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-locus-sashimi-d}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-locus-sashimi-z}
    \end{subfigure}
	\caption[Constant-region isoforms in \textit{X. maculatus}]{\textbf{Constant-region isoforms in \textit{X. maculatus}:} Coverage and sashimi plots of STAR-aligned RNA-seq reads from \textit{X. maculatus} samples, demonstrating the splicing behaviour of IGH1 constant-region isoforms. (A) IGHM exon splicing, showing alternative splicing patterns of IGHM-TM and IGHM-S; (B) IGHD exon splicing, including splicing of \cm{1} to \cd{1}; (C) IGHZ exon splicing.}
	\label{fig:xma-locus-sashimi}
	\end{figure}

	
	\subsection{Variable regions}
	
	In total, 99 VH segments, 9 DH segments and 14 JH segments were identified in the \textit{X. maculatus} IGH locus. In all cases, the great majority of these segments appear to exclusively serve IGHM and IGHD and exclude IGHZ; as in most teleost IGH loci, the IGHZ constant region in \textit{X. maculatus} has its own dedicated DH and JH regions (IGHDZ01 and IGHJZ01) separate from those serving IGHM and IGHD; % TODO: Compare these to the orphaned D/J segments in TK to test vestigial IGHZ hypothesis
in addition, unlike in most teleost loci, the \textit{X. maculatus} IGHZ constant region is upstream of the majority of VH segments in the locus, with only a single VH segment between the start of the locus and IGHDZ01. This lone VH segment, IGHV01-01, is distinct in sequence from the other VH segments in the locus (\Cref{fig:xma-vh-families-tree,fig:xma-vh-families-map}, second from left), but appears to be functional; its status as the only VH segment upstream of IGHZ indicates that the potential sequence diversity available to this isotype is very limited, with only a single VDJ combination available to the recombination process. In this case, any and all antigen-binding diversity in \textit{Xiphophorus} \textit{IGHZ} must arise from junctional diversity alone. 

	Unlike the IGHZ variable region, the VH region downstream of IGHZ but upstream of IGHM is highly expanded, with 98 identified VH segments -- more than four times the total number seen in \textit{N. furzeri}, and almost six times the number in the largest \textit{N. furzeri} sublocus. At least 74 of these VH segments appear to be fully in-frame and functional and equipped with functional RSS sequences, representing one of the highest numbers of functional VH segments seen in any teleost IGH locus; it remains to be seen whether IGHV01-01 is also capable of recombining with DH segments downstream of the IGHZ constant region, and so constitutes part of the range of VDJ combinations available to IGHM and IGHD. The VH sequences in the \textit{X. maculatus} locus are much more tightly packed than in the \textit{N. furzeri} locus, consistent with a lower overall prevalence of repetitive regions (21\%) in the \textit{X. maculatus} genome. However, the final four VH segments (of which two are structurally intact) are downstream of the last identified DH segment and inverted relative to the rest of the locus, and may not be able to successfully undergo VDJ recombination. 
	
	In total, the VH regions in \textit{X. maculatus} \textit{IGH} fall into 22 families, of which six contain multiple segments; strikingly, the single VH segment serving IGHZ (IGHV01-01) represents a distinct family which is distinct from any other segment in the locus \Cref{fig:xma-vh-families-tree}. \dots % TODO: Phylogenetic analysis of VH in xma zs nfu
	
	In addition to its highly expanded VH region, the variable region of the \textit{X. maculatus} locus is unusual in the arrangement of its DH and (to a lesser extent) its JH regions. Rather than a single, relatively densely-packed D-region, the \textit{X. maculatus} locus contains DH segments interspersed throughout the long stretch of the locus between the end of the \textit{IGHZ} constant region and the start of that of IGHM. Many of these lone DH segments are closely associated with single JH segments, raising the possibility of a more cluster-like behaviour in which each $V_n-D_{1-2}-J_1$ group acts as a distinct recombination unit. However, the presence of a dense eight-segment J-region immediately upstream of IGHM suggests a more conventional translocon behaviour; it remains to be seen which of these traditional models of antigen-receptor structure more closely matches the \textit{in vivo} recombination behaviour of this locus.
	
	% TODO: Discuss here phylogenetic relationship between Nfu and Xma DH and (esp) JH segments (esp. eight segments in main JH block)
	
	Finally, as is the case with \textit{N. furzeri}, the recombination signal sequences (RSSs) in \textit{X. maculatus} IGH correspond closely to the standard expectations across the vertebrates, with the expected heptamer and nonamer consensus sequences (\Cref{fig:xma-rss-seqlogo-all-heptamer,fig:xma-rss-seqlogo-all-nonamer}) and spacer length distributions (\Cref{fig:xma-rss-seqlogo-all-spacer}, 95\% of RSS spacers within \bp{1} of the expected conserved length).
	

% TODO: compare VH families from platy and TK, align C-exons between TK and platy (and other cyprinodontiforms)
	
	
	\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{_Figures/png/xma-vh-families-tree}
	\caption[Dendrogram of VH families in the in \textit{X. maculatus} \textit{IgH} locus]{\textbf{Dendrogram of VH families in the in \textit{X. maculatus} (\textit{IgH}) locus:} Dendrogram of sequence similarity of VH segments in the \textit{X. maculatus} locus, arranged by single-linkage clustering on nucleotide sequence identity. The red line indicates the 80\% cutoff point for family assignment, while branch colour indicates family membership:  VH families containing multiple segments are uniquely coloured, while single-segment families are in grey.}
	\label{fig:xma-vh-families-tree}
	\end{figure}
	
	\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{_Figures/png/xma-vh-families-map}
	\caption[Heatmap of VH families in the in \textit{X. maculatus} \textit{IgH} locus]{\textbf{Heatmap of VH families in the in \textit{X. maculatus} (\textit{IgH}) locus:} Heatmap of family relationships among \textit{X. maculatus} VH segments, with coloured shading indicating families and red dots indicating pairwise nucleotide sequence identity of at least 80\%. VH families containing multiple segments are uniquely coloured, while single-segment families are in grey.}
	\label{fig:xma-vh-families-map}
	\end{figure}
	
	\begin{figure}
		\begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-rss-seqlogo-all-heptamer}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-rss-seqlogo-all-spacer}
    \end{subfigure}
    \begin{subfigure}{0em}
        \phantomsubcaption{}
        \label{fig:xma-rss-seqlogo-all-nonamer}
    \end{subfigure}
	\includegraphics[width=\textwidth]{_Figures/png/xma-rss-seqlogo-all}
	\caption[Recombination signal sequences in the \textit{X. maculatus} \textit{IGH} locus]{\textbf{Recombination signal sequences in the \textit{X. maculatus} \textit{IGH} locus:} (A) Sequence composition of conserved heptamer sequences across all \textit{X. maculatus} heavy-chain RSSs; (B) length distribution of unconserved spacer sequences in \textit{X. maculatus} heavy-chain RSSs; (C) sequence composition of conserved heptamer sequences across all \textit{X. maculatus} heavy-chain RSSs.}
	\label{fig:xma-rss-seqlogo-all}
	\end{figure}
	% TODO: Add in supplementary figure of heptamer/nonamer sequences separated by segment type

	%\subsection{Discussion} % Do I need this?	


\section{Comparitive analysis of \textit{IgH} constant-region structure in the Cyprinodontiformes}
\label{sec:comparative-loci}

	\subsection{Evolutionary history of IgZ}
	
	\subsection{Sublocus multiplicity and orientation}
	
	\subsection{Exon usage in transmembrane IgM \& other isoforms}






%\section{The combination of genome and BAC data reveals a two-part sublocus on chromosome 6}
%
%The structure of the immunoglobulin heavy chain (IgH) locus provides the fundamental underpinning of antibody diversity in a species, defining the space of possible sequences arising from the VDJ recombination process, the functional antibody classes available via the constant-region exons, and the possible splicing behaviour of those exons. In order to investigate the dynamics of antibody immunity in a species, it is therefore necessary to have at least a working knowledge of its IgH structure, and ideally a comprehensive characterisation. Such a characterisation provides both the primer sequences required for immunoglobulin sequencing and the complete V, D and J-segment databases required for analysis of Ig-seq data, as well as directly providing information about antibody immunity in a species.
%
%In order to obtain such a characterisation for the turquoise killifish, I sequenced and assembled several  clones from the TK bacterial artificial chromosome (BAC) library suspected to contain parts of the locus. (see \dots for details on the BAC assembly method). In parallel, I collated comprehensive VH, JH and CH-exon sequence databases for three model teleost species with previously assembled IgH loci: zebrafish (\textit{Danio rerio} \citep{} ), stickleback (\textit{Gasterosteus aculeatus} \citep{}) and medaka (\textit{Oryzias latipes} \citep{} ). These databases were aligned against the most recent assembly of the TK genome (\dots) to identify scaffolds potentially bearing part of the IgH locus. Finally, the BAC inserts and genome scaffolds were aligned together to produce a single, contiguous locus sequence.
%
%In total, one chromosome (chr6) and four smaller scaffolds were identified as bearing probable IgH locus fragments (\Cref{tab:nfu_locus_scaffolds}). The existing chromosome 6 assembly contained what appeared to be a complete (V-D-J-CM-CD) sublocus on the sense strand, as well as a single downstream VH segment in antisense; the presence and position of this inverted V suggested the presence of at least one more sublocus, present in antisense downstream of the already-assembled sublocus, which had not been included in the chromosome assembly. % Other identified scaffolds
%
%Taken together, these data suggested a model of the TK IgH locus in which two apparently-complete subloci face each other on opposite strands of chromosome 6. This hypothesis was confirmed by the BAC assembly data, which confirmed the existence of at least two non-overlapping subloci with distinct VH structures. The first of these, from the aligned inserts of several BACs including \dots, shows high identity with the 5' sense-strand sublocus present on chromosome 6, while the second, contained in the inserts of BACs \dots, aligns with scaffolds \dots and the inverted VH region on chromosome 6, with the \dots insert extending into the post-IgH 3' part of the chromosome. % Figure for this?
%Integrating all these data gave rise to the contiguous locus sequence outlined in \dots. %figure
%
%As is the case in most teleosts, but unlike in salmonids, only a single IgH locus could be identified; the two genome scaffolds that could not be integrated into the locus sequence contained either VH and JH or VH sequences only, and appear to represent orphaned variable-region segments rather than a functional second locus. No other chromosome was found to contain a multi-part, contiguous sequence resembling that found on chromosome 6. 
%
%In conclusion, the African turquoise killifish genome contains a single immunoglobulin heavy-chain gene locus, approximately 305 kilobases in length. This sequence appears to contain exactly two structurally-complete subloci, present in tandem but on opposite strands, each containing VH, DH (see below), JH and constant-region sequences. The structural completeness of both subloci suggest that both may be capable of recombining to produce functional IgH transcripts; \dots % Functionality of antisense sublocus?
%
%% NB: It would be really nice to get single-cell DNA sequencing of killifish B-cells and see if there are any cells where both subloci are recombined; do you maybe get more than one chance per chromosome?
%
%\begin{table}
%\begin{threeparttable}
%\begin{tabular}{ccccccc}\toprule
%Scaffold & Total length (kb) & V & J & \cm{} & \cd{} & \cz{} & Included in locus?\\
%chr6 & 6195.6 & 15 & 7 & 5 & 11 & Yes\\
%scf10901 & 1.4 & 0 & 0 & 0 & 3 & Yes\\
%scf35954 & 16.3 & 3 & 0 & 0 & 0 & No\\
%scf36277 & 18.9 & 2 & 1 & 0 & 0 & No\\
%scf9157 & 7.2 & 0 & 7 & 4 & 0 & Yes\\
%\end{tabular}
%\caption{Summary of genome scaffolds containing putative \textit{IgH} locus fragments}
%\end{threeparttable}
%\label{tab:nfu_locus_scaffolds}
%\end{table}
%
%\begin{figure}
%\caption{Assembly of the \textit{N. furzeri} \textit{IgH} locus sequence from genome scaffolds and BAC inserts.} 
%\end{figure}
%
%\section{}
%
%% Similarity between IGH1 and IGH2 (esp D, J, constant regions - but not V?)
%
%Having generated a contiguous sequence for the IgH locus, constant-region exons from the three reference species were realigned to the new locus in the same manner they had been aligned to the genome and BAC sequences previously. These alignments were used as the basis for detailed characterisation of CH exons in the TK locus. To refine intron/exon boundaries and analyse splicing behaviour, RNA-seq data from eight mature killifish (GRZ-Bellemans, 4 $\times$ 6-week-old fish, 4 $\times$ 16-week-old) was mapped to the locus with STAR (\dots methods link here, inc. repeat-masking \dots); the combination of a B-cell-rich source tissue (\dots citation needed \dots) and the use of data from several individuals enabled clear delineation of intron/exon boundaries, as well as identification of the second transmembrane exon for both IgM and IgD in both subloci. % Example figure?
%
%The exon structure of both subloci is identical, though there are some differences in spatial arrangement: in IgH2, for example, the 5' and 3' IgD exons are separated by a large \dots region which is absent in IgH1. The exon sequences of the two subloci is also very similar, with an average of \dots \% sequence identity between corresponding exons. % Does this tell us anything about the duplication timing?
%Given this high level of similarity in sequence and exon arrangement, it is perhaps unsurprising that the splice patterns of the two subloci appear to be identical % Though, many reads will be mapping to both
