\section{Immunoglobulin sequencing of turquoise killifish samples}

\subsection{Husbandry and sample processing}
\label{sec:methods:igseq:samples}

For experiments 1 \& 2, 32 male turquoise killifish from strain GRZ-AD were hatched in a single cohort and raised in % single housing from ... weeks of age.
Husbandry conditions were in accordance with % TODO: Read JoVE paper
; briefly, ...

See APPENDIX for a complete list of fish used in this experiment.

Groups of fish from this cohort were sacrificed at intervals (see TABLE) by anaesthetisation in 1.5g/L Tricaine killing solution followed by flash freezing in liquid nitrogen. The sacrificed fish were homogenised in a mortar and pestle filled with liquid nitrogen, and the resulting powder was transferred to \ml{50} conical tubes for storage at \degC{-80}. The distribution of body weights at death of each cohort are shown in FIGURE. % Should this last sentence be in results?

Following homogenisation, total RNA was extracted from the samples using QIAzol (DETAILS); the detailed extraction protocol is as follows...

The concentration of the total RNA was tested with ..., and the quality was verified using the Agilent TapeStation 4200...; only RNA samples with an RNA Integrity Number equivalent (RINe) of at least 8.0 were used for library preparation % is this true?

\subsection{Library preparation and sequencing}

Total RNA from killifish samples (see \autoref{sec:methods:igseq:samples}) 

For each library to be sequenced, \ng{750} killifish total RNA was combined with \ul{2} \umol{10} gene-specific primer (GSP) corresponding to the \textit{IgH} isotype(s) to be sequenced (see TABLE), along with nuclease-free water to a total volume of \ul{8}. The RNA-primer mixture was incubated for 2 minutes at \degC{70} to facilitate primer annealing, then held at \degC{42} % in accordance with the protocol given in...

\ul{8} of the annealed RNA-primer mixture was then combined with \ul{12} of the reverse-transcription master-mix, prepared as follows:

\begin{itemize}
\item \ul{2} SMARTScribe reverse transcriptase (\uul{100}, DETAILS)
\item \ul{4} SMARTScribe first-strand buffer (\x{5}, DETAILS)
\item \ul{2} SmartNNNa barcoded template-switch adapter (\umol{10}, BOX for DETAILS)
\item \ul{2} DTT (\mmol{20}, DETAILS, ABBREV)
\item \ul{2} dNTP mix (\umol{10} each, DETAILS)
\item \ul{0.5} RNasin RNase inhibitor (\uul{40}, DETAILS) 
\end{itemize}

The prepared reverse-transcription mixture was incubated for \hr{1} at \degC{42} for the reverse-transcription reaction, then mixed with \ul{1} of uracil DNA glycosylase (\uul{5}, DETAILS) and incubated for a further \min{40} at \degC{37} to digest residual TSA oligonucleotides. The reaction product was purified using SeraSure beads at \x{0.7} concentration, eluting in \ul{16.5} clean elution buffer. % Solutions and buffers section?

Following cleanup, the cDNA sequences were made double-stranded and amplified using nested PCR with Kapa DNA polymerase, % HotStart PCR ReadyMix?
(TABLE, PCR 1; primer details in TABLE and FIGURE), then re-purified with SeraSure beads (\x{0.7} input, resuspend in \ul{25} EB, \mins{5} elution). Partial Illumina adaptors (without indices for multiplexing) were added in a second PCR (TABLE, PCR 2; primer details in TABLE and FIGURE), followed by a further SeraSure bead cleanup (0.7x = 17.5uL input, resuspend in 15uL EB, 5 mins elution). Finally, a third PCR with Illumina TruSeq adaptor primers (...) was used to add a unique index combination to each library in a given experiment (see TABLE for index data for ...); this was followed by a further bead cleanup (...) prior to library quality control.

After the final bead cleanup, the concentration of each library was tested with the Qubit (details), while the size distribution was examined using the Agilent TapeStation 4200; this information together enabled the estimation of the concentration of the desired library band (at LENGTHS) for each sample. The samples from each experiment were pooled together according to these estimates, such that every library had the same estimated concentration in the pooled sample. 


\begin{table}
\begin{threeparttable}
\caption{Gene-specific primers (GSPs) for reverse-transcription of IgH sequences from total turquoise killifish RNA.}
\begin{tabular}{cccc} %TODO : Get proper names for primers
\textbf{Primer Name} & \textbf{Sequence} & Isotype specificity & \textbf{Target \textit{IgH} exon}\\
... & .. & IgM-TM, IgM-S, IgD & CM1\\
... & .. & IgM-TM, IgM-S & CM2\\
% ...
\end{tabular}
%TODO Add annotations re shared exons
\end{threeparttable}
\label{tab:lp_gsps}
\end{table}