\section{Immunoglobulin sequencing of turquoise killifish samples}

\subsection{Husbandry and sample processing}

\subsection{Library preparation and sequencing}

\subsection{Ig-seq data analysis}

\subsubsection{Data pre-processing with pRESTO}

Sequencing data were uploaded by the sequencing provider to Illumina BaseSpace, having already been demultiplexed and trimmed of adaptors (see TABLE for indices used in each experiment). In experiment 2, the number of reads mapped to index combinations other than those used in library preparation, as a percentage of total reads, was used to measure the level of contamination produced using library preparation (TABLE, FIGURE). From BaseSpace, the raw data were downloaded to a local file server using \texttt{basemount}, from which they were uploaded to the high-performance computing cluster of the MPI Age.

Pre-processing of the raw data into repertoire sequences was done using the \texttt{pRESTO} toolkit [CITE]. The pre-processing pipeline used in the analysis is displayed graphically in FIGURE and can be summarised as follows:

\begin{table}
\newcounter{presto}
\setcounter{presto}{1}
\begin{threeparttable}
\begin{tabular}{cc>{\texttt}cl}
Stage & Name & \texttt{pRESTO} script CITE & Description\\
\arabic{presto}\stepcounter{presto} & Quality trimming & FilterSeq quality & Discard reads with a mean PHRED quality score below VALUE (corresponding to an error probability of VALUE).\\
\arabic{presto}\stepcounter{presto} & Primer masking & MaskPrimers score & Trim invariant primer sequences from appropriate ends of 3' and 5' reads.\\
\arabic{presto}\stepcounter{presto} & UMI extraction & MaskPrimers score & Trim remaining TSA sequence from 5' read and extract 16bp UMI sequence.\\ % TODO: quality? error levels?
\arabic{presto}\stepcounter{presto} & UMI clustering & ClusterSets barcode & Group similar (but not necessarily identical) UMIs together as potential barcode errors.\\
\arabic{presto}\stepcounter{presto} & Split UMI clusters & ClusterSets set & Split UMI clusters with dissimilar insert sequences as potential barcode collisions (or mis-clusterings from stage STAGE).\\ % TODO: Details of clustering methods (cd-hit-est, vsearch)
\arabic{presto}\stepcounter{presto} & Unify clusters & PairSeq & Copy UMI and cluster annotations from 5' reads to their 3' pairs.\\
\arabic{presto}\stepcounter{presto} & Build consensus sequences & BuildConsensus & Group sequences with matching cluster assignments and infer consensus sequence of each group.\\ % TODO: Details of consensus building?
\arabic{presto}\stepcounter{presto} & Unify MIGs & PairSeq & Match consensus annotations across read pairs and discard consensus sequences without a pair.\\
\arabic{presto}\stepcounter{presto} & Merge sequence pairs & AssemblePairs sequential & Align and merge paired consensus sequences to generate complete repertoire sequences.\\
\arabic{presto}\stepcounter{presto} & Collapse sequences & CollapseSeq & Group identical repertoire sequences with different cluster IDs together and record abundance of each sequence.\\ % These are assumed to represent distinct RNA molecules of the same sequence present in the original sample
\arabic{presto}\stepcounter{presto} & Discard singletons & SplitSeq group & Discard repertoire sequences represented by only one read across all cluster IDs.\\
\end{tabular}
\end{threeparttable}
\end{table} % TODO: Table or list?

% TODO: Discuss rarefaction

At the completion of this pipeline, each sequence in the resulting FASTA file represents a single sequence present in the original RNA sample, and is annotated with an estimate of the true abundance of that sequence.