

\section{Characterising killifish \textit{IgH} loci using genomic sequence data}

\subsection{Identifying locus scaffolds}

V segments, J segments and constant-region exons from reference sequences (see \autoref{sec:locus_ref}) were aligned to the genome assembly of % species being used for de novo characterisation
using BLASTN (for nucleotide sequences) and TBLASTN (for protein sequences). Alignments were filtered based on E-value and the number of different reference sequences that were aligned to a given region. Scaffolds from the genome that:

\begin{enumerate}
\item[a] contained strong alignments to multiple different locus sequence types (e.g. V & J segments, or different constant-region exons); or
\item[b] were covered by segment alignments constituting at least a threshold amount (1\%) of their total length
\end{enumerate}

were retained as putative locus scaffolds. Of these, scaffolds whose aligned locus regions constituted less than 1\% of their total length (i.e. that fail to fulfill condition (b) above) were truncated to within 200kb on either side of the terminal segment alignments, on the assumption that these long scaffolds contained a locus rather than forming part of one. The resulting set of scaffolds was used as the basis for all downstream locus characterisation methods for that species.

\subsection{Identifying locus segments from cross-species sequence alignment}

The cross-species alignments used to identify putative genome scaffolds in ...

\subsection{Identifying locus segments from probabilistic sequence matching}

RSS sequence databases from reference species were aligned together with the multiple-sequence-alignment program PRANK % or another program, insert here
with a high gap-opening penalty of $x$ %...
. % Separate long and short RSS alignments?
The resulting alignment file was used with the Hidden-Markov-Model-inference program hmmer to 

\subsection{Identifying locus segments from RNA-seq mapping}


